JobsPortal/src/main/java/com/ps/js/controller/JobsController.java

```diff
@@ -0,0 +1,363 @@
+			skillPayload.setSkillId(skill.getSkillId());
+			skillPayload.setSkillName(skill.getSkillName());
+			listSkillPayload.add(skillPayload);
+		}
+		System.out.println("listSkillPayload :: " + listSkillPayload);
+		return new ResponseEntity<List<SkillPayload>>(listSkillPayload, HttpStatus.OK);
+	}
+
+	/**
+	 * To find all the job location set for open a new job
+	 * 
+	 * @return {@link ResponseEntity}
+	 */
+	@GetMapping("/find-all-locations")
+	public ResponseEntity<List<JobLocationPayload>> findAllJobLocation() {
+		JobLocationPayload jobLocationPayload = null;
+		List<JobLocationPayload> listJobLocationPayload = new ArrayList<JobLocationPayload>();
+		List<JobLocation> listJobLocations = jobLocationServices.fetchAllLocation();
+		for (JobLocation location : listJobLocations) {
+			jobLocationPayload = new JobLocationPayload();
+			jobLocationPayload.setLocationId(location.getLocationId());
+			jobLocationPayload.setLocationName(location.getLocationName());
+			listJobLocationPayload.add(jobLocationPayload);
+		}
+
+		return new ResponseEntity<List<JobLocationPayload>>(listJobLocationPayload, HttpStatus.OK);
+	}
+
+	/**
+	 * To find all roles and responsebility for open a new job
+	 * 
+	 * @return {@link ResponseEntity}
+	 */
+	@GetMapping("/find-all-roles-and-responsebility")
+	public ResponseEntity<List<RolesAndResponsebilityPayload>> findAllRolesAndResponsebility() {
+		RolesAndResponsebilityPayload rolesAndResponsebilityPayload = null;
+		List<RolesAndResponsebilityPayload> listRolesAndResponsebilityPayload = new ArrayList<RolesAndResponsebilityPayload>();
+		List<RolesAndResponsebility> listRolesAndResponsebility = rolesAndResponsebilityService
+				.fetchAllRolesAndResponsebility();
+		for (RolesAndResponsebility rolesAndResponsebility : listRolesAndResponsebility) {
+			rolesAndResponsebilityPayload = new RolesAndResponsebilityPayload();
+			rolesAndResponsebilityPayload.setId(rolesAndResponsebility.getId());
+			rolesAndResponsebilityPayload.setRolesResponsebility(rolesAndResponsebility.getRolesResponsebility());
+			listRolesAndResponsebilityPayload.add(rolesAndResponsebilityPayload);
+		}
+
+		return new ResponseEntity<List<RolesAndResponsebilityPayload>>(listRolesAndResponsebilityPayload,
+				HttpStatus.OK);
+	}
+
+	/**
+	 * For delete jobDetails by using job-code and return confirmation for the
+	 * deleted record
+	 * 
+	 * @param jobCode
+	 * @return {@link ResponseEntity}
+	 */
+
+	@DeleteMapping("/delete-job-details/{job-code}")
+	public ResponseEntity<String> deleteJobDetails(@PathVariable("job-code") String jobCode) {
+		System.out.println(jobCode);
+		JobDetails jobDetails = jobDetailsService.fetchJobDetailsByJobCode(jobCode).get();
+
+		return new ResponseEntity<String>("Job details Delete :" + jobDetails.getJobId(), HttpStatus.OK);
+	}
+
+	/**
+	 * To find all the record based on job status
+	 * 
+	 * @param jobStatus
+	 * @return {@link ResponseEntity}
+	 */
+	@GetMapping("/find-job-by-job-status/{job-status}")
```

Comment:
URI design is not good, we need to have a end point to find a job, and followed by a filter criteria to handle all the scenarios. 
Basically one end point should get jobs, based on all the possible filters